packages/2d/src/components/Latex.ts

```diff
@@ -0,0 +1,81 @@
+import {mathjax} from 'mathjax-full/js/mathjax';
+import {TeX} from 'mathjax-full/js/input/tex';
+import {SVG} from 'mathjax-full/js/output/svg';
+import {AllPackages} from 'mathjax-full/js/input/tex/AllPackages';
+import {liteAdaptor} from 'mathjax-full/js/adaptors/liteAdaptor';
+import {RegisterHTMLHandler} from 'mathjax-full/js/handlers/html';
+import {computed, initial, signal} from '../decorators';
+import {Image, ImageProps} from './Image';
+import {
+  DependencyContext,
+  SignalValue,
+  SimpleSignal,
+} from '@motion-canvas/core/lib/signals';
+import {OptionList} from 'mathjax-full/js/util/Options';
+
+const adaptor = liteAdaptor();
+RegisterHTMLHandler(adaptor);
+
+const jaxDocument = mathjax.document('', {
+  InputJax: new TeX({packages: AllPackages}),
+  OutputJax: new SVG({fontCache: 'local'}),
+});
+
+export interface LatexProps extends ImageProps {
+  tex?: SignalValue<string>;
+  renderProps?: SignalValue<OptionList>;
+}
+
+export class Latex extends Image {
+  private static latexPool: Record<string, HTMLImageElement> = {};
+
+  @initial({})
+  @signal()
+  public declare readonly renderProps: SimpleSignal<OptionList, this>;
+
+  @signal()
+  public declare readonly tex: SimpleSignal<string, this>;
+
+  public constructor(props: LatexProps) {
+    super(props);
+  }
+
+  @computed()
+  protected override image(): HTMLImageElement {
+    // Render props may change the look of the TeX, so we need to cache both
+    // source and render props together.
+    const src = `${this.tex()}::${JSON.stringify(this.renderProps())}`;
+    if (Latex.latexPool[src]) {
+      return Latex.latexPool[src];
+    }
+
+    // Convert to TeX, look for any errors
+    const tex = this.tex();
+    const svg = adaptor.innerHTML(jaxDocument.convert(tex, this.renderProps()));
+    if (svg.includes('data-mjx-error')) {
+      const errors = svg.match(/data-mjx-error="(.*?)"/);
+      if (errors && errors.length > 0) {
+        throw new Error(`Invalid MathJax: ${errors[1]}`);
+      }
+    }
+
+    // Encode to raw base64 image format
+    const text = `data:image/svg+xml;base64,${btoa(
+      `<?xml version="1.0" encoding="UTF-8" standalone="no" ?>\n${svg}`,
+    )}`;
+    const image = document.createElement('img');
+
+    image.src = text;
+    if (!image.complete) {
+      DependencyContext.collectPromise(
```

Comment:
I was seeing the image come back as not complete which is why I left this in. I’ll debug it and double check that we need this.

Reply:
I'm seeing some weirdness here; if I don't `yield add` the component, it shows up inconsistently. If I do, it shows up consistently regardless of if I have the async functionality here or not. I'm going to add that to the example and remove the async code, but see if any adjustments to how I load the generated tex changes anything in that regard.

Reply:
Doesn't look like we get those, but we can reuse the image element for sure. I'll keep the `yield add` in the example.

Reply:
I checked on my side and you're right - the image is not ready immediately, so this is still necessary:
```ts
image.src = text;
if (!image.complete) {
  DependencyContext.collectPromise(
    new Promise((resolve, reject) => {
      image.addEventListener('load', resolve);
      image.addEventListener('error', reject);
    }),
  );
}
```
The reason doing `yield add` works is because it causes the animation to do `Promise.all` on an empty array, which - even though it doesn't await anything - returns back control:
```ts
async function test() {
  console.log('1');
  await Promise.all([]);
  console.log('3');
}
  
test();
console.log('2');
```
This tiny pause seems to be enough for the image to become ready, but I'm not certain if for bigger equations on slower machines, it would still work. So collecting the promise wouldn't hurt.